File: 5-Patron/php/Hechizo.php
<?php
include './LibroVentisca.php';
include './LibroLluviaEspadas.php';
include './LibroFortaleza.php';
include './Animacion.php';
include './Sprite.php';
include './Particulas.php';

class Hechizo extends HabilidadMagica{
	private $particulas;
	private $animacion;
	private $sprite;
	private $libro;
	private $ataque;


	public function __construct($nivel, $descripcion,$particulas,$animacion,$sprite,$libro,$ataque){
    parent::__construct($nivel, $descripcion);
    $this->particulas = $particulas;
    $this->libro = $libro;
    $this->animacion = $animacion;
    $this->sprite = $sprite;
    $this->ataque = $ataque;
	}

	public function getNivel(){
		return parent::getNivel();	
	}

	public function setNivel($nivel){
		parent::setNivel($nivel);
	}

	public function getDescripcion(){
		return parent::getDescripcion();	
	}

	public function setDescripcion($descripcion){
		parent::setDescripcion($descripcion);
	}

	public function getParticulas(){
		return $this->particulas;	
	}

	public function setParticulas($particulas){
		$this->particulas = $particulas;
	}

	public function getAnimacion(){
		return $this->animacion;	
	}

	public function setAnimacion($animacion){
		$this->animacion = $animacion;
	}

	public function getSprite(){
		return $this->sprite;	
	}

	public function setSprite($sprite){
		$this->sprite = $sprite;
	}

	public function getLibro(){
		return $this->libro;	
	}

	public function setLibro($libro){
		$this->libro = $sprite;
	}

	public function getAtaque(){
		return $this->ataque;	
	}

	public function setAtaque($ataque){
		$this->ataque = $ataque;
	}

	public function verInfoHabilidad(){
    return  parent::verInfoHabilidad();
  	}

  	public function  subirNivelHabilidad(){
  		return  parent::subirNivelHabilidad();
  	}

  	public function verStats(){
    return  "Ataque: " . $this->ataque .
            "Libro con: " . $this->libro->verStats();
  	}

  	public function crearLibro($tipoLibro,$duracion,$texto){
		if ($tipoLibro == 1) {
        	$this->libro = new LibroFortaleza($duracion,$texto);
        	return $this->libro;
    	} elseif ($tipoLibro == 2) {
        	$this->libro = new LibroVentisca($duracion,$texto);
        	return $this->libro;
    	}elseif ($tipoLibro == 3) {
    		$this->libro = new LibroLluviaEspadas($duracion,$texto);
    		return $this->libro;
    	}else{
    		return null;
    	}
	}
}
?>

File: 5-Patron/Java/Hechizo.java
package patron5;

public class Hechizo extends HabilidadMagica{
    private Particulas particulas;
    private Libro libro;
    private int ataque;
    private Sprite sprite;
    private Animacion animacion;

    public Hechizo(Particulas particulas, Libro libro, int ataque, Sprite sprite, Animacion animacion,  int nivel, String descripcion) {
        super(nivel, descripcion);
        this.particulas = particulas;
        this.libro = libro;
        this.ataque =  ataque;
        this.sprite = sprite;
        this.animacion = animacion;
    }

    

    
  
    public Animacion getAnimacion() {
        return animacion;
    }

    public int getAtaque() {
        return ataque;
    }

    public Libro getLibro() {
        return libro;
    }

    public Particulas getParticulas() {
        return particulas;
    }

    public Sprite getSprite() {
        return sprite;
    }

    public void setAnimacion(Animacion animacion) {
        this.animacion = animacion;
    }

    public void setAtaque(int ataque) {
        this.ataque = ataque;
    }

    public void setLibro(Libro libro) {
        this.libro = libro;
    }

    public void setParticulas(Particulas particulas) {
        this.particulas = particulas;
    }

    public void setSprite(Sprite sprite) {
        this.sprite = sprite;
    }

    @Override
    public void subirNivelHabilidad() {
        super.subirNivelHabilidad(); 
    }
    
   
    public String verStats(){
        return("Ataque: "+this.ataque+"Libro con: "+ this.libro.verStats());
    }
    
    public Libro crearLibro(int tipoLibro,int duracion, String texto){
        switch (tipoLibro) {
            case 1:
                this.libro = new LibroFortaleza(duracion, texto);
                break;
            case 2:
                this.libro = new LibroVentisca(duracion, texto);
                break;
            case 3:
                this.libro = new LibroLluviaEspadas(duracion, texto);
                break;
            default:
                break;
        }
        return null;
    } 
}
